angular/projects/admin/src/app/admin/form/textbox-rich/textbox-rich.component.ts

```diff
@@ -131,11 +133,42 @@ export class TextboxRichComponent implements MatFormFieldControl<string>, Contro
   }
 
   setDisabledState(isDisabled: boolean): void {
-    this.formEditor.isReadOnly = isDisabled;
+    // this.formEditor.isReadOnly = isDisabled;
   }
 
-  onChange({ editor }: ChangeEvent) {
-    this._onChangeCallback(editor.getData());
+  onChange(editor) {
+    this._onChangeCallback(editor);
     this.stateChanges.next();
   }
+
+  onEditorChange(event) {
+    console.log('onEditorChange', event);
+  }
+
+  onFileUploadRequest(event) {
+    console.log('onFileUploadRequest', event);
+  }
+
+  onFileUploadResponse(event) {
+    console.log('onFileUploadResponse', event);
+  }
+
+  insert_image (event) {
+    this._zone.run(() => {
+      const dialogRef = this.dialog.open(FilePickerDialogComponent, {width: '800px'});
+      const subscription = dialogRef.afterClosed().subscribe(file => {
+        if (!!file) {
+          event.insertHtml(`<img src=/_/file/${file.id}?s=medium />`, 'unfiltered_html');
+          // For localhost development -->
+          // this.siteLink$.subscribe(site => {
+            // event.insertHtml(`<img src=https://${site}/_/file/${file.id}?s=medium />`, 'unfiltered_html');
+          // })
```

Comment:
You can remove this comment. Because you can add "localhost support" to a site by listing `localhost` in the list of that site's domains.

Reply:
Resolved on this commit https://github.com/oddbit/tanam/pull/266/commits/80992f5e27f46e340199dacbbba71d676674b93a